### PR TITLE
[Oracle] Add network disruption handling

### PR DESCRIPTION
**Pull request by nenadnoveljic**
### What does this PR do?

Adding network disruption handling. The error message is parsed, and in the case of network disruption (`connection refused`) the Agent logs basic diagnostic steps.

### Motivation

In the case of a network disruption on the client site, the Agent won't be able to get the metrics from the database. The native error message might not be clear to everyone, which could result in unnecessary support tickets - we can't do anything in such a case. The goal is to emphasize that the underlying error is in the network infrastructure, and point the user to some basic diagnostic steps.

https://datadoghq.atlassian.net/browse/DBMON-2781

### Describe how to test/QA your changes

Misconfigure the port and observe if the new message appears in the log file.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/collector/corechecks/oracle-dbm/sql_wrappers.go [hunk 1]
```diff
@@ -72,6 +76,26 @@ func isConnectionError(err error) bool {
...
+	if err == nil {
+		return false, err
+	}
+	if isConnectionRefused(err) {
+		closeDatabase(c, db)
+		return true, fmt.Errorf(`%w
+The network connection between the Agent and the database server is disrupted. Run one of the following commands on the machine where the Agent is running to test the network connection: 
+nc -v dbserver port
+telnet dbserver port
+curl dbserver:port`,
```
Comment (jmeunier28) [inline pkg/collector/corechecks/oracle-dbm/sql_wrappers.go:93]: @alexandre-normand wondering if you think its worth hooking this up to agent warnings 

that way we could create a troubleshooting doc for oracle, [similar to sqlserver](https://docs.datadoghq.com/database_monitoring/setup_sql_server/troubleshooting/#common-connection-issues), which might look nicer than printing these commands in the agent logs

**Review by jmeunier28 — APPROVED** [review]
seems OK for now 👍 

I think if we want to do more complicated in-agent troubleshooting help for this integration in the future, we will want to use agent warnings. That way we can link customers to public troubleshooting docs && display which agents aren't working in the setup issues page.

Screen Shot 2023-09-13 at 9 33 49 AM

**Review by maycmlee — APPROVED** [review]
Just a small suggestion, but approving.

File: pkg/collector/corechecks/oracle-dbm/sql_wrappers.go (hunk 1 above)
Comment (nenadnoveljic) [reply pkg/collector/corechecks/oracle-dbm/sql_wrappers.go:93]: I created the ticket https://datadoghq.atlassian.net/browse/DBMON-2796 to do that. Let me know your final decision.

**Comment by nenadnoveljic** [comment]
> seems OK for now 👍
> 
> I think if we want to do more complicated in-agent troubleshooting help for this integration in the future, we will want to use agent warnings. That way we can link customers to public troubleshooting docs && display which agents aren't working in the setup issues page.
> 
> Screen Shot 2023-09-13 at 9 33 49 AM

How are these warning generated? With log.Warn?

**Comment by nenadnoveljic** [comment]
> Just a small suggestion, but approving.

Thx!

**Comment by alexandre-normand** [comment]
> > seems OK for now 👍
> > I think if we want to do more complicated in-agent troubleshooting help for this integration in the future, we will want to use agent warnings. That way we can link customers to public troubleshooting docs && display which agents aren't working in the setup issues page.
> > Screen Shot 2023-09-13 at 9 33 49 AM
> 
> How are these warning generated? With log.Warn?

That's a good question. I think we have to sort of reverse engineer how the [check.warning](https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/postgres.py#L862-L868) works in the python integrations but it looks like it might just end up setting warnings on the check and returning them via [GetWarnings](https://github.com/DataDog/datadog-agent/blob/main/pkg/collector/check/check.go#L36). Note that there's a format to our warning messages to properly identify an error (code, troubleshooting docs url). You can see an example [here](https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/postgres.py#L568-L577).

All that said, I do agree this is the way to go for something like this. Warnings should show up in the agent status if recorded that way and would also make it to datadog for us to display in the UI. The increased visibility is pretty useful.

**Comment by nenadnoveljic** [comment]
> > > seems OK for now 👍
> > > I think if we want to do more complicated in-agent troubleshooting help for this integration in the future, we will want to use agent warnings. That way we can link customers to public troubleshooting docs && display which agents aren't working in the setup issues page.
> > > Screen Shot 2023-09-13 at 9 33 49 AM
> > 
> > 
> > How are these warning generated? With log.Warn?
> 
> That's a good question. I think we have to sort of reverse engineer how the [check.warning](https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/postgres.py#L862-L868) works in the python integrations but it looks like it might just end up setting warnings on the check and returning them via [GetWarnings](https://github.com/DataDog/datadog-agent/blob/main/pkg/collector/check/check.go#L36). Note that there's a format to our warning messages to properly identify an error (code, troubleshooting docs url). You can see an example [here](https://github.com/DataDog/integrations-core/blob/master/postgres/datadog_checks/postgres/postgres.py#L568-L577).
> 
> All that said, I do agree this is the way to go for something like this. Warnings should show up in the agent status if recorded that way and would also make it to datadog for us to display in the UI. The increased visibility is pretty useful.

I opened a separate ticket for instrumenting the Agent with the warnings: https://datadoghq.atlassian.net/browse/DBMON-2797

**Pull request merged**